File: src/main/java/upbrella/be/user/entity/BlackList.java
package upbrella.be.user.entity;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Getter
@Entity
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class BlackList {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private long socialId;

    public static BlackList createNewBlackList(long socialId) {

        return BlackList.builder()
                .socialId(socialId)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/user/entity/BlackList.kt
package upbrella.be.user.entity

import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
class BlackList(
    val socialId: Long,
    val blockedAt: LocalDateTime,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
) {
    companion object {
        @JvmStatic
        fun createNewBlackList(socialId: Long, blockedAt: LocalDateTime = LocalDateTime.now()): BlackList {
            return BlackList(
                socialId = socialId,
                blockedAt = blockedAt,
            )
        }
    }
}